File: C#/IsAnagram.cs
//O(n)
//O(1)
public class Solution
{
    public bool IsAnagram(string s, string t)
    {
        if (s.Length != t.Length) return false;

        int[] charCount = new int[256];

        for (int i = 0; i < s.Length; i++)
        {
            charCount[s[i]]++;
            charCount[t[i]]--;
        }

        foreach (int count in charCount)
        {
            if (count != 0) return false;
        }

        return true;
    }
}

File: C++/isAnagram.cpp
// Time: O(N)
// Space: O(N)

#include <string>
#include <unordered_map>

class Solution
{
public:
    bool isAnagram(std::string s, std::string t)
    {

        std::unordered_map<char, int> word1;
        std::unordered_map<char, int> word2;

        for (char c : s)
        {
            word1[c]++;
        }

        for (char c : t)
        {
            word2[c]++;
        }

        return word1 == word2;
    }
};